test/integration/minimal_scene.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+#include <QtTest/QtTest>
+#include <ignition/common/Console.hh>
+#include <ignition/math/Color.hh>
+#include <ignition/math/Pose3.hh>
+#include <ignition/rendering/Camera.hh>
+#include <ignition/rendering/RenderEngine.hh>
+#include <ignition/rendering/RenderingIface.hh>
+#include <ignition/rendering/Scene.hh>
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "../helpers/TestHelper.hh"
+#include "ignition/gui/Application.hh"
+#include "ignition/gui/GuiEvents.hh"
+#include "ignition/gui/Plugin.hh"
+#include "ignition/gui/MainWindow.hh"
+
+int g_argc = 1;
+char **g_argv = new char *[g_argc];
```

Comment:
```suggestion
char **g_argv = {"minimal_test"};
```

Reply:
I'm on Bionic and the test passes for me. It also passes on [GH actions](https://github.com/ignitionrobotics/ign-gui/pull/221/checks?check_run_id=2913506543). Does it fail locally for you?

The proposed change won't compile. This does:

```suggestion
char *g_argv[] = {"unused"};
```

But it triggers a warning:

```
warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
```

I think we could suppress that warning, but I'm not sure that the change is needed at all.

Reply:
I created a PR here https://github.com/ignitionrobotics/ign-gui/pull/244

According with the official [Qt documentation](https://doc.qt.io/qt-5/qguiapplication.html#QGuiApplication).

```
Warning: The data referred to by argc and argv must stay valid for the entire lifetime of the QGuiApplication object.
In addition, argc must be greater than zero and argv must contain at least one valid character string.
```